### DISCUSSION TITLE
A couple of Compute Settings questions.

### DISCUSSION THREAD

**Discussion by brknsoul** [vladmandic/sdnext#993] [Q&A] [answered]
_(AMD RX 6600 XT 8Gb, Windows 10, direct-ml)_

Regarding --no-half/--no-half-vae, and Enable Upcast Sampling; do I need all these checked? Which is faster?

And regarding Cross-attention optimization method; Which is fastest?

**Comment by vladmandic** [comment, marked as answer]
`directml` does not have autocast capabilities on their own, so we've implemented those manually.  
result is that fp16 can be used instead of fp32 and that is much faster.
but side-effect is that fp16 is still quite a bit sensitive and can cause bad quality images in some cases.

so for speed don't enable any.
but if you have quality issues, try enabling one-by-one.

regarding cross-attention, sdp with directml works, but its very memory hungry. perhaps sub-quad?
